### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Vitalii-Kazymyrovych**

### CODE REVIEW

File: src/main/java/com/project/bookstore/mapper/RoleMapper.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.project.bookstore.mapper;
+
+import com.project.bookstore.config.MapperConfig;
+import com.project.bookstore.dto.role.RoleDto;
+import com.project.bookstore.model.Role;
+import com.project.bookstore.dto.role.CreateRoleRequestDto;
+import org.mapstruct.*;
```
Comment (MentorMA) [inline src/main/java/com/project/bookstore/mapper/RoleMapper.java]: Don`t use "*" in imports

File: src/main/java/com/project/bookstore/BookStoreApplication.java [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+    private final RoleService roleService;
 
     public static void main(String[] args) {
         SpringApplication.run(BookStoreApplication.class, args);
     }
+
+    @Override
+    public void run(String... args) throws Exception {
+        roleService.fillRolesTable();
+    }
```
Comment (MentorMA) [inline src/main/java/com/project/bookstore/BookStoreApplication.java]: Why do you do it here? You can fill it by liquibase

**Review by MentorMA — CHANGES_REQUESTED** [review]

File: src/main/java/com/project/bookstore/BookStoreApplication.java (hunk 2 above)
Comment (Vitalii-Kazymyrovych) [reply src/main/java/com/project/bookstore/BookStoreApplication.java]: Want to have more flexibility in application.

File: pom.xml [hunk 3]
```diff
@@ -38,6 +38,11 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-security</artifactId>
+			<version>3.2.2</version>
```
Comment (Rommelua) [inline pom.xml]:
```suggestion
```

File: src/main/java/com/project/bookstore/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+@RequiredArgsConstructor
+public class UserController {
+    private final UserService userService;
+
+    @PreAuthorize("hasAuthority('admin')")
+    @PutMapping("/update_roles")
```
Comment (Rommelua) [inline src/main/java/com/project/bookstore/controller/UserController.java]:
```suggestion
    @PutMapping("/update-roles")

```

File: src/main/java/com/project/bookstore/dto/user/UpdateUserRolesRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package com.project.bookstore.dto.user;
+
+import jakarta.validation.constraints.Email;
+import java.util.List;
+
+public record UpdateUserRolesRequestDto(
+        @Email String email,
```
Comment (Rommelua) [inline src/main/java/com/project/bookstore/dto/user/UpdateUserRolesRequestDto.java]: use userId

File: src/main/java/com/project/bookstore/BookStoreApplication.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/com/project/bookstore/BookStoreApplication.java]: This is hardcode, not a flexibility )
Delete this method and all related code, please

File: src/main/java/com/project/bookstore/config/SecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(
+                        auth -> auth
+                                .requestMatchers("/auth/**")
+                                .permitAll()
+                                .anyRequest()
+                                .authenticated()
+                )
+                .httpBasic(Customizer.withDefaults())
```
Comment (koval-kn) [inline src/main/java/com/project/bookstore/config/SecurityConfig.java:38]: static method import is more preferred
```suggestion
                .httpBasic(withDefaults())
```

**Review by koval-kn — APPROVED** [review]

**Pull request merged**